fabfile.py

```diff
@@ -1,197 +1,518 @@
+    command = "sudo -u postgres psql -lqt | cut -d \| -f 1 | grep -w {database_name} | wc -l"
+    database_exists = "0" not in local(
+        command.format(database_name=some_env.database_name), capture=True
+    )
+
+    if database_exists:
+        postgres_command("DROP DATABASE {0}".format(some_env.database_name))
+
+    postgres_command("CREATE DATABASE {0}".format(some_env.database_name))
+    postgres_command("GRANT ALL PRIVILEGES ON DATABASE {0} TO {1}".format(
+        some_env.database_name, DB_USER
+    ))
+
+
+def postgres_load_database(backup_name, new_env):
+    if not os.path.isfile(backup_name):
+        raise ValueError("unable to find a backup at {}".format(
+            backup_name)
+        )
+
+    local("sudo -u postgres psql -d {0} -f {1}".format(
+        new_env.database_name,
+        backup_name
+    ))
+
+
+def services_symlink_nginx(new_env):
+    abs_address = "{}/etc/nginx.conf".format(new_env.project_directory, "")
+    if not os.path.isfile(abs_address):
+        raise ValueError(
+            "we expect an nginx conf to exist at {}".format(abs_address)
         )
-        local("git push {0} {1}:master".format(
-            remote_name, current_branch_name)
+
+    symlink_name = '/etc/nginx/sites-enabled/{}'.format(MODULE_NAME)
+    if os.path.islink(symlink_name):
+        local("sudo rm {}".format(symlink_name))
+
+    local('sudo ln -s {0} {1}'.format(abs_address, symlink_name))
+
+
+def services_symlink_upstart(new_env):
+    abs_address = "{}/etc/upstart.conf".format(new_env.project_directory, "")
+    if not os.path.isfile(abs_address):
+        raise ValueError(
+            "we expect an upstart conf to exist {}".format(abs_address)
         )
+    symlink_name = '/etc/init/{}.conf'.format(PROJECT_NAME)
+    local("sudo rm -f {}".format(symlink_name))
 
+    local('sudo ln -s {0} {1}'.format(
+        abs_address,
+        symlink_name
+    ))
 
-@task
-def create_heroku_instance(name, username):
+
+def services_create_local_settings(new_env, additional_settings):
+    new_settings = copy.copy(additional_settings)
+    new_settings["db_name"] = new_env.database_name
+    new_settings["db_user"] = DB_USER
+    template = jinja_env.get_template(
+        'conf_templates/local_settings.py.jinja2'
+    )
+    output = template.render(new_settings)
+    local_settings_file = '{0}/{1}/local_settings.py'.format(
+        new_env.project_directory, MODULE_NAME
+    )
+
+    local("rm -f {}".format(local_settings_file))
+
+    with open(local_settings_file, 'w') as f:
+        f.write(output)
+
+
+def services_create_gunicorn_conf(new_env):
+    template = jinja_env.get_template('conf_templates/gunicorn.conf.jinja2')
+    output = template.render(
+        env_name=new_env.virtual_env_path
+    )
+    gunicorn_conf = '{0}/etc/gunicorn.conf'.format(
+        new_env.project_directory
+    )
+
+    local("rm -f {}".format(gunicorn_conf))
```

Comment:
If this actually finds something, at least I'd want a log message?
I suspect maybe moving to e.g. ./etc/gunicorn.conf.bak might be worth doing? (I'd want to know what it was and if it had changed I imagine)